rest/route/host_create.go

```diff
@@ -70,7 +70,38 @@ func (h *hostCreateHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	ids := []string{}
-	for i := 0; i < numHosts; i++ {
+	t, err := task.FindOneId(h.taskID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding task '%s'", h.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
+		})
+	}
+	initialHosts, err := host.FindHostIntentsByTask(ctx, h.taskID, t.Execution)
```

Comment:
@Kimchelly this line is erroring only when I call the route and I'm not really sure why because the function seems to be working fine when I call it in the test by itself?

Reply:
https://parsley.mongodb.com/test/evergreen_ubuntu2204_test_rest_route_patch_88c719945ddf70e235db7ea9765e101c479261a1_66b2aa9fda7edd000728b633_24_08_06_22_58_41/0/e43c926c744887dd001376a709e7bb6c?bookmarks=0,4270&shareLine=849

Not the exact same test but this has the same issue as this log

Reply:
Can you try [this](https://github.com/evergreen-ci/evergreen/pull/8164#discussion_r1709831215)? It should log the error from the response when the test fails.